server/extensions/exchange-rate/caching.js

```diff
@@ -0,0 +1,39 @@
+const NodeCache = require('node-cache');
+const axios = require('axios');
+const { DEFAULT_CURRENCY, CAD } = require('../common/config/constants/currency.constants');
+
+const FRANKFURTER_API_URL = 'https://api.frankfurter.app';
+const CACHE_TIME = 86400;
+
+const currencyRateCache = new NodeCache({
+  stdTTL: CACHE_TIME,
+});
+
+const fetchExchangeRate = async () => {
+  try {
+    const response = await axios.get(
+      `${FRANKFURTER_API_URL}/latest?base=${DEFAULT_CURRENCY}&symbols=${CAD}`
+    );
+    const exchangeRate = response.data.rates.CAD;
+
+    currencyRateCache.set('exchangeRate', exchangeRate);
```

Comment:
Should return an object with the exchange rates for better extension {"CAD": 1.3}
```suggestion
    currencyRateCache.set('exchangeRate', exchangeRate);
```